File: 2.Algorithm/99.ETC/숫자 블록/숫자 블록.py
def p(n):
    for i in range(2, int(n**0.5)):
        if n % i == 0 and n//i <= 10000000:
            return n//i
    return 1

def solution(begin, end):
    answer = []
    for i in range(begin, end+1):
        answer.append(p(i))
    return answer

File: 2.Algorithm/99.ETC/숫자 블록.java
import java.util.ArrayList;

class Solution {
    
    public ArrayList<Long> solution(long begin, long end) {
        ArrayList<Long> answer = new ArrayList<Long>();
        for(long i = begin; i < end+1; i++){
            answer.add(p(i));
        }
        return answer;
    }
    
    public long p(long n){
        for (long i = 2; i < Math.sqrt(n); i++){
            if (n % i == 0 && n/i <= 10000000){
                return n/i;
            }
        }
        return 1;
    }
}